### PR TITLE
JENKINS-64130 JENKINS-64255 Tables to divs compat

### PR DESCRIPTION
**Pull request by FarmGeek4Life**
Tested on my instance, this change fixed all issues I was seeing with "Configure Global Security"

### CODE REVIEW

**Comment by FarmGeek4Life** [comment]
Changes were a near copy-and-paste following the pull request fixing [Promoted Builds Plugin](https://github.com/jenkinsci/promoted-builds-plugin/pull/149)

**Comment by FarmGeek4Life** [comment]
I'm not sure on the cause of the test failures, unless it is specific to the linux build; the tests that failed do pass on my system running Windows 10.

File: src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly [hunk 1]
```diff
@@ -1,16 +1,16 @@
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:a="/lib/active_directory">
   <j:choose>
     <j:when test="${descriptor.canDoNativeAuth()}">
       <!-- code path for ActiveDirectoryAuthenticationProvider: running on Windows -->
       <f:entry><!-- get some indentation -->
-        <table style="width:100%">
+        <div style="width:100%">
```
Comment (timja) [inline src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly:7]:
```suggestion
        <a:blockWrapper>
```

File: src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly [hunk 2]
```diff
@@ -27,23 +27,23 @@
...
+                </a:blockWrapper>
                 <div align="right">
                   <input type="button" value="Delete Domain" class="repeatable-delete" style="margin-left: 1em;" />
                 </div>
               </f:repeatable>
             </f:entry>
             <st:include page="configAdvanced.jelly" class="${descriptor.clazz}"/>
           </f:optionalBlock>
-        </table>
+        </div>
```
Comment (timja) [inline src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly:38]:
```suggestion
        </a:blockWrapper>
```

File: src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly [hunk 3]
```diff
@@ -27,23 +27,23 @@
...
         <st:include page="configCache.jelly" class="${descriptor.clazz}"/>
       </f:entry>
     </j:when>
     <j:otherwise>
       <!-- code path for ActiveDirectoryUnixAuthenticationProvider: running on Unix -->
       <f:entry field="domains" title="${%Domains}">
         <f:repeatable field="domains" add="${%Add Domain}">
-          <table width="100%">
+          <div width="100%">
```
Comment (timja) [inline src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly:46]:
```suggestion
          <a:blockWrapper>
```

File: src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly [hunk 4]
```diff
@@ -66,7 +66,7 @@
             <f:nested>
               <f:validateButton with="name,servers,site,bindName,bindPassword,tlsConfiguration" title="${%Test Domain}" method="validateTest"/>
             </f:nested>
-          </table>
+          </div>
```
Comment (timja) [inline src/main/resources/hudson/plugins/active_directory/ActiveDirectorySecurityRealm/config.jelly:69]:
```suggestion
          </a:blockWrapper>
```

File: src/main/resources/lib/active_directory/blockWrapper.jelly [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <st:documentation>
+    This adds a wrapper allowing adding an ID to a field or group of fields that can be targeted by JavaScript
+    The wrapper will be a `table` tag before form layout changes are merged: https://github.com/jenkinsci/jenkins/pull/3895,
```
Comment (timja) [inline src/main/resources/lib/active_directory/blockWrapper.jelly:5]:
```suggestion
    The wrapper will be a `table` tag before 2.264,
```

**Review by timja — COMMENTED** [review]
few fixups, thanks for this

**Comment by FarmGeek4Life** [comment]
It looked great that time until the build timed out.

**Comment by timja** [comment]
> It looked great that time until the build timed out.

I've re-triggered it

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
@batmat @rsandell @MRamonLeon @varyvol @fcojfernandez could someone take a look please?

**Review by fqueiruga — APPROVED** [review]

**Review by MRamonLeon — APPROVED** [review]
pretty clean solution :heart_eyes:

**Comment by batmat** [comment]
2.22 out with this fix. 

Thanks!

**Comment by batmat** [comment]
Released in 2.22. Thanks a bunch @FarmGeek4Life !

**Pull request merged**